packages/maskbook/src/plugins/Trader/pipes.ts

```diff
@@ -36,6 +38,8 @@ export function resolveDataProviderLink(dataProvider: DataProvider) {
             return 'https://www.coingecko.com/'
         case DataProvider.COIN_MARKET_CAP:
             return 'https://coinmarketcap.com/'
+        case DataProvider.UNISWAP:
```

Comment:
use table driven.

```ts
const providers: Record<DataProvider, string> = {
    [DataProvider.COIN_GECKO]: '... url ...'
}
return providers[dataProvider]
```

Reply:
@zhouhanseng If you decide to refactor this to table-driven please take `EnumRecord` a look in the codebase.